erpnext/stock/doctype/purchase_receipt/purchase_receipt.py

```diff
@@ -192,7 +192,7 @@ def get_gl_entries(self, warehouse_account=None):
 					gl_entries.append(self.get_gl_dict({
 						"account": warehouse_account[d.warehouse]["account"],
 						"against": stock_rbnb,
-						"cost_center": d.cost_center,
+						"cost_center": d.cost_center or erpnext.get_default_cost_center(self.company),
```

Comment:
Why setting default cost center here? Instead, in validate we should set default cost center if missing.